pkgs/tools/admin/yandexcloud-cli/default.nix

```
@@ -0,0 +1,106 @@
+{ lib
+, stdenv
+, fetchurl
+, makeBinaryWrapper
+, installShellFiles
+, testers
+  # update script
+, writeShellScript
+, curl
+, common-updater-scripts
+}:
+let
+  version = "0.108.1";
+  hashes = {
+    "aarch64-darwin" = "sha256-WCOVUB0IxEarclKPHj8qds6sBUutBr/+mNG/etV5Igs=";
+    "aarch64-linux" = "sha256-c8zhW4rD+bMhSGEA0YLyah6chRIrPpPO9sbswrgQCFg=";
+    "i686-linux" = "sha256-Y1YhGf7asOB0IjncTKfevpM+Z36hwOKesPEbSeINTsI=";
+    "x86_64-darwin" = "sha256-mK6GRfntgnJ5bKKn5PWFda59c2ZA9S4EgHjBtVG+73Q=";
+    "x86_64-linux" = "sha256-a7KQopidzzqePG9MSm68xQ8CqTtJrFIi5lDSt8EswB4=";
+  };
+
+  storageURL = "https://storage.yandexcloud.net/yandexcloud-yc/release";
+
+  fetchSource = system: hash:
+    let
+      throwSystem = throw "Unsupported system: ${system}";
+      platform = lib.systems.elaborate system;
+      arch =
+        if platform.isAarch64 then "arm64"
+        else if platform.isx86_64 then "amd64"
+        else if platform.isx86_32 then "386"
+        else throwSystem;
+      os =
+        if platform.isLinux then "linux"
+        else if platform.isDarwin then "darwin"
+        else throwSystem;
+    in
+    fetchurl {
+      url = "${storageURL}/${version}/${os}/${arch}/yc";
+      inherit hash;
+      executable = true;
+    };
+in
+stdenv.mkDerivation (finalAttrs:
+let
+  # For generating shell completion even when cross-compiling. Note that
+  # binaries are statically linked so we can execute them without patching.
+  nativeBin = finalAttrs.passthru.sources.${stdenv.buildPlatform.system} or null;
+in
+{
+  pname = "yandexcloud-cli";
+  inherit version;
+
+  src = finalAttrs.passthru.sources.${stdenv.hostPlatform.system};
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeBinaryWrapper ] ++ lib.optional (nativeBin != null) installShellFiles;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp -a $src $out/bin/yc
+  '' + lib.optionalString (nativeBin != null) ''
```

```suggestion
  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
```
normally we solve this by not generating the completion file or using buildPackages.yandexcloud-cli